modules/ServerAPI/src/server/routes/ThingServiceRoutes.coffee

```diff
@@ -285,6 +287,42 @@ updateThing = (thing, testMode, callback) ->
 		)
 
 
+exports.bulkPostThingsSaveFile = (req, resp) ->
+	exports.bulkPostThingsInternal req.body, (response) =>
+		checkFilesAndUpdate = (thing, callback) ->
+			fileVals = serverUtilityFunctions.getFileValuesFromEntity thing, false
+			filesToSave = fileVals.length
+
+			completeThingUpdate = (thingToUpdate)->
+				updateThing thingToUpdate, false, (updatedThing) ->
+					callback updatedThing, 200
+
+			fileSaveCompleted = (passed) ->
+				if !passed
+					callback "file move failed", 500
+				if --filesToSave == 0 then completeThingUpdate(thing)
+
+			if filesToSave > 0
+				prefix = serverUtilityFunctions.getPrefixFromEntityCode thing.codeName
+				for fv in fileVals
+					csUtilities.relocateEntityFile fv, prefix, thing.codeName, fileSaveCompleted
```

Comment:
Is `fv` being modified by this function, and the change flowing all the way back to `thing`?

Reply:
relocate entity file is modifying the thing and then calling its call back "fileSaveCompleted".  This way, the customer specific server function is responsible for doing something with the file.

Reply:
I guess I am just confused that this `relocateEntityFile` function is not having the `thing` variable passed to it. How does it access the `thing`? Does it look it up by codeName?

Reply:
its getting passed just the file value `fv`. I just added some more code comments to clarify (I had to look it up to).